classes/UI/PagesComponent.php

```diff
@@ -34,68 +54,73 @@ class PagesComponent implements ComponentInterface, MutatableOptionsInterface
-			'limit' => ['type' => 'int', 'title' => lmsg('thankyou.component.options.num_items'), 'default' => 10, 'min' => 1, 'max' => 50],
-			'user_id' => ['type' => 'int', 'title' => lmsg('thankyou.component.options.user_id'), 'default' => 0, 'input' => 'user_picker', 'width' => 'medium'],
+			'title'          => ['type' => 'string', 'title' => ($this->lmsg)('thankyou.component.options.custom_title'), 'default' => '', 'placeholder' => ($this->lmsg)('thankyou.component_heading')],
+			'show_header'    => ['type' => 'bool', 'title' => ($this->lmsg)('thankyou.component.options.show_header'), 'default' => true, 'mutate_on_change' => true],
+			'allow_new'      => ['type' => 'bool', 'default' => true, 'title' => ($this->lmsg)('thankyou.component.options.show_button')],
+			'profile_images' => ['type' => 'bool', 'default' => false, 'title' => ($this->lmsg)('thankyou.component.options.profile_images')],
+			'comments'       => ['type' => 'bool', 'default' => false, 'title' => ($this->lmsg)('common.show_comments')],
+			'user_id'        => ['type' => 'int', 'title' => ($this->lmsg)('thankyou.component.options.user_id'), 'default' => null, 'input' => 'user_picker', 'width' => 'medium'],
+			'group_ids'      => ['type' => 'array_int', 'title' => ($this->lmsg)('thankyou.common.filter_by_groups'), 'default' => [], 'input' => 'group_picker', 'width' => 'medium'],
+			'limit'          => ['type' => 'int', 'title' => ($this->lmsg)('thankyou.component.options.num_items'), 'default' => 10, 'min' => 1, 'max' => 50]
 		];
 	}
 
 	/**
 	 * Render this component with the specified options
 	 *
-	 * @param string $id_string
+	 * @param string           $id_string
 	 * @param OptionsInterface $options
-	 * @param Application $app
+	 * @param Application      $app
 	 *
 	 * @return string
 	 */
 	public function ShowBody($id_string, OptionsInterface $options, Application $app)
 	{
-		$args = array();
-
-		/** @var ThanksRepository $repository */
-		$repository = $app['thankyou.repository'];
-
-		$user_id = $options->Get('user_id');
-		$limit = $options->Get('limit');
-
-		if ($user_id)
-			$thanks = $repository->GetForUser($user_id, $limit);
-		else
-			$thanks = $repository->GetRecent($limit);
-
 		/**
-		 * @var ThanksListView $view
+		 * @var ThankYous $api
 		 */
-		$view = $app['thankyou.thanks_list_view'];
-		$args['items.datasrc'] = $view->Show($thanks, [
-			'profile_images' => $options->Get('profile_images')
-		], $app->security);
+		$api = $app[ThankYous::class];
+
+		$thank_user_id = $options->Get('user_id');
+		$group_ids     = $options->Get('group_ids');
 
-		if (count($args['items.datasrc']) === 0)
+		$args = [];
+
+		$args['ty_list.limit'] = $options->Get('limit');
+		if (!is_int($args['ty_list.limit']))
 		{
-			$msg = lmsg('thankyou.component.no_thanks_all');
+			throw new ComponentRuntimeException("Failed to Show Thank You Component, Limit is not an integer");
```

Comment:
I'd recommend just using the default value instead of throwing an exception. Exceptions can be quite destructive in Pages components.

Reply:
Fixed in 47734e831787f98ff8860969073d1755c9601b01